python-package/lightgbm/sklearn.py

```diff
@@ -1067,6 +1067,11 @@ def feature_name_(self) -> List[str]:
         if not self.__sklearn_is_fitted__():
             raise LGBMNotFittedError('No feature_name found. Need to call fit beforehand.')
         return self._Booster.feature_name()  # type: ignore[union-attr]
+    
+    @property
+    def feature_names_in_(self) -> List[str]:
+        """:obj:`list` of shape = [n_features]: Sklearn-style property for feature names."""
```

Comment:
Please update this with the following:

* remove "sklearn-style property for" and instead just say what it is, something like "names for features"
* this should only be available in a fitted model, right? If so, please guard it like this:

https://github.com/microsoft/LightGBM/blob/cc733f8595267f313886f92ed5d1285010ba8f3f/python-package/lightgbm/sklearn.py#L993-L994

* explain in the docs what will happen when accessing this attribute if you never provided feature names (e.g. just passed raw `numpy` arrays as training data)